lib/linter/linter.js

```diff
@@ -1408,12 +1394,14 @@ class Linter {
             ? getDirectiveComments(sourceCode, ruleId => getRule(slots, ruleId), options.warnInlineConfig, config)
             : { configuredRules: {}, enabledGlobals: {}, exportedVariables: {}, problems: [], disableDirectives: [] };
 
-        // augment global scope with declared global variables
-        addDeclaredGlobals(
-            sourceCode.scopeManager.scopes[0],
-            configuredGlobals,
-            { exportedVariables: commentDirectives.exportedVariables, enabledGlobals: commentDirectives.enabledGlobals }
-        );
+        // augment global scope with declared global variables for ESTree only
+        if (sourceCode.isESTree) {
+            addDeclaredGlobals(
+                sourceCode.scopeManager.scopes[0],
+                configuredGlobals,
+                { exportedVariables: commentDirectives.exportedVariables, enabledGlobals: commentDirectives.enabledGlobals }
+            );
+        }
```

Comment:
Similar to https://github.com/eslint/eslint/pull/18448#discussion_r1609865923.

I think we should rather update the tests that are failing without this change to return a valid AST.

Reply:
Now that the tests are fixed, we could revert this change, and also revert the change above https://github.com/eslint/eslint/pull/18448#discussion_r1620551343?